subsys/pm/device.c

```diff
@@ -44,6 +58,16 @@ int pm_device_state_set(const struct device *dev,
 		}
 
 		action = PM_DEVICE_ACTION_SUSPEND;
+
+		/*
+		 * Power domain will be OFF, now notify all dependents
+		 * about it
+		 */
+		if (domain) {
+			(void)device_supported_foreach(dev,
+				      domain_supported_cb,
+				      (void *)PM_DEVICE_ACTION_POWER_DOMAIN_OFF);
```

Comment:
I wonder if devices should enable notification (via a flag), since many probably don't care. cc: @JordanYates ?

Reply:
to avoid the function call ? if they don't want they can ignore (don't handle) the action. Just wondering if it is not better to keep it simpler.

Reply:
I agree with @ceolin, explicitly ignoring the action in the callback is simpler than adding a reason for the callback to not be run.